mcs/class/corlib/Test/System/ArrayTest.cs

```diff
@@ -3385,6 +3385,30 @@ public void IEnumerator_Current_Reset ()
 		}
 	}
 
+	[Test]
+	public void IEnumerator_Dispose ()
+	{
+		IEnumerable<int> e = new int[] { 1 };
+		var en = e.GetEnumerator ();
+		Assert.IsTrue (en.MoveNext (), "#1");
+		Assert.IsFalse (en.MoveNext (), "#2");
+		en.Dispose ();
+		Assert.IsFalse (en.MoveNext (), "#3");
+	}
+
+	[Test]
+	public void IEnumerator_ZeroSize ()
+	{
+		IEnumerable<int> e = Array.Empty<int> ();
+		var en = e.GetEnumerator ();
+		Assert.IsFalse (en.MoveNext (), "#1");
+
+		e = Array.Empty<int> ();
+		en = e.GetEnumerator ();
+		Assert.IsFalse (en.MoveNext (), "#2");
+	}
```

Comment:
this test passes before the fix already for me, is that expected?

Reply:
yes, just added it for the test coverage

Reply:
Only IEnumerator_Dispose fails with master